File: src/main/java/com/example/backend/model/AuthUser.java
package com.example.backend.model;

import com.example.backend.entity.Role;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AuthUser {
    private int id;
    private String name;
    private String surname;
    private String email;
    private Role role;

}

File: src/main/ng/frontend/src/app/model/authuser.ts
import {Role} from "./role";

export class AuthUserModel {
    id: string;
    name: string;
    surname: string;
    email: string;
    status: boolean;
    role: Role;

    static cloneBase(user: AuthUserModel) : AuthUserModel{
        const cloneUserModel: AuthUserModel = new AuthUserModel();
        cloneUserModel.id = user.id;
        cloneUserModel.email=user.email;
        cloneUserModel.name=user.name;
        cloneUserModel.surname=user.surname;
        cloneUserModel.status=user.status;
        cloneUserModel.role=user.role;
        return cloneUserModel;
    }
}